### ISSUE TITLE
`&` breaks JSON

### ISSUE THREAD

**Issue by aarikpokras** [turrnut/simas#4]
I'll make a PR soon, but in program.js, line 13 where it makes a JSON object out of the return value of `simas.run()`, I ran test.simas and it threw an error. I would say maybe try to keep it a string.

**Comment by turrnut** [comment]
Did you try compiling the file using `compiler.js` first? I have separated the compiler and the runtime now so you might need to compile your `.simas` file to a `.csa` file, which stands for Compiled Simple Assembly. And then you can run the `.csa` file using `program.js`.

**Comment by aarikpokras** [comment]
Oh, I see. Sorry about that.
Would it be:
```
node compiler.js test.simas
```
And then I run the csa file?

**Comment by aarikpokras** [comment]
Also, just curious, how were you able to make it so that it compiles?

**Comment by turrnut** [comment]
Yes. After that you would do
`node program.js test.csa`
Also the `compiler.js` basically saves the instructions and operands in a file in the format of encrypted json.

**Comment by aarikpokras** [comment]
How do you encrypt and decrypt the JSON? I'm curious, this is really cool.

**Comment by turrnut** [comment]
I used the `xString` function in the file `common.js` file to both encrypt and decrypt the JSON. `xString` performs XOR(Exclusive Or) on every single character in the JSON string with the key 71 which is just a random number I made up. Now this function would return encrypted text. However, if you feed the encrypted text into the `xString` function again it will return the original text, thus decrypting it. This is achieved through the nature of XOR—where `a^b^b` will always equal `a`
